pkg/controller/daemon/controller.go

```diff
@@ -304,9 +304,32 @@ func (dsc *DaemonSetsController) deletePod(obj interface{}) {
 	}
 }
 
+// checkDsNodeMatch returns true when node label matches the daemon set nodeSelector
+// and ds has no specific nodeName or ds's specific nodeName matches the name of the node
+func checkDsNodeMatch(ds *extensions.DaemonSet, node *api.Node) bool {
+	nodeSelector := labels.Set(ds.Spec.Template.Spec.NodeSelector).AsSelector()
+	shouldRun := nodeSelector.Matches(labels.Set(node.Labels))
+	// If the daemon set specifies a node name, check that it matches with nodeName.
+	nodeName := node.Name
+	shouldRun = shouldRun && (ds.Spec.Template.Spec.NodeName == "" || ds.Spec.Template.Spec.NodeName == nodeName)
+	return shouldRun
+}
+
 func (dsc *DaemonSetsController) addNode(obj interface{}) {
 	// TODO: it'd be nice to pass a hint with these enqueues, so that each ds would only examine the added node (unless it has other work to do, too).
-	dsc.enqueueAllDaemonSets()
+	dsList, err := dsc.dsStore.List()
+	if err != nil {
+		glog.Errorf("Error enqueueing daemon sets: %v", err)
```

Comment:
This will only error if there are no daemonsets to list. This is very common (any cluster that has daemonset resource enabled but no deamonsets created yet). Please don't error log and give it a high info log.